src/components/checkbox/switch.ts

```diff
@@ -36,38 +36,37 @@ export default class IgcSwitchComponent extends IgcCheckboxBaseComponent {
 
   private handleClick() {
     this.checked = !this.checked;
+    this.emitEvent('igcChange', { detail: this.checked });
```

Comment:
By moving the event emitter here, we will stop emitting when changing the value via keyboard.